### ISSUE TITLE
[Feature Request] Improve bookmarks branching

### ISSUE THREAD

**Issue by kakhulu31** [Cohee1207/SillyTavern#363]
**Have you searched for similar [requests](https://github.com/Cohee1207/SillyTavern/issues?q=)?**
Yes

**Is your feature request related to a problem? Please describe.**
It's not a problem, just inconvenience. When you create a new bookmark, 1. it blocks you from choosing a different swipe from the latest message (unless you move the latest message down) 2. it is always placed at the very end of the conversation.

**Describe the solution you'd like**
A perfect solution would be much harder since it would require implementing a completely new tree/branching chat system. 
But to improve the current bookmarks system I think some changes can be made. To address mentioned above issues:
1st issue. Place bookmark before the desired message BUT cover the desired message as well i.e:
`
1 message
2 message
(bookmark gets created here and includes 3rd message)
3 message (we create bookmark after this message)
 `
2nd issue. Bookmark button should be placed inside the edit message menu, so that user could create a new bookmark after any existing message in the chat, and not just after the latest one.

![image](https://github.com/Cohee1207/SillyTavern/assets/51547727/d4b735af-b6d8-4a28-b6c9-36395a8812a0)

**Describe alternatives you've considered**
Tree/branching chat system presumably proposed here https://github.com/Cohee1207/SillyTavern/issues/353

**Comment by kakhulu31** [comment]
Yes I noticed. It looks more accurate, but now I can't see how to delete it. 
And the 2nd issue (inability to create bookmarks after any existing message in the chat) which is much more important to me is yet to be solved, I guess?
Also the button to stop generation (for streamed messages) disappeared for some reason.
![image](https://github.com/Cohee1207/SillyTavern/assets/51547727/f0feb720-291d-4648-a5c4-ef36118263a3)

**Comment by kakhulu31** [comment]
Oh and another thing. Character importing doesn't work now.

**Comment by morphles** [comment]
Ok, so just to add, about dialog tree thing (what I was intending to create issue for/looking for similar), since it was mentioned here. I think it would be extremely nice to have dialog tree implemented. Sometimes I click regen, and get shittier result, then can't go back to what I had. Or maybe I get few good results across some regens. And then it would be very interesting to explore more than one path. I'm thinking of something similar to what board game (well, chess and go at least) analysis tools have.

**Comment by morphles** [comment]
How do I access it? :) Then it is quite well hidden.

**Comment by morphles** [comment]
But actually, it seems more limited. Or again, not super clear. Say I want to go 10 messages back, and provide different response/input, to have whole new branch. Also seems it's just for last message. Well or I'll need to spend time experimenting and familiarizing with it.

**Comment by Cohee1207** [comment]
Check out this extension. You can also set a bookmark on any message now.
https://github.com/city-unit/SillyTavern-Timelines
